### PR TITLE
Added testcase files for Sort, SortByColumns functions

### PR DESCRIPTION
**Pull request by mitugm14**
Please review the files and suggest if any of the scenarios need to be updated.

Thanks,
Mitu

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/Sort.txt [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+
+>> Sort(["a", "b", "D", "x", "J", Error({Kind: 11})], Value)
+#Error(Kind=Validation)
+
+>> Sort([1/0], Blank())
+#Error(Kind=Div0)
+
+// Sort error with bool value as sort formula
+>> Sort([1/0],true)
+#Error(Kind=Div0)
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/Sort.txt:113]: This is not correct - the result is a valid table with a single record with an error property)

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/Sort.txt [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+  { First: "Alice", Age: 5, Properties: { Vaccinated: true, HasDog: true }},
+  { First: "John", Age: 17, Properties: { Vaccinated: false, HasDog: false }},
+  { First: "Emily", Age: 29, Properties: { Vaccinated: true, HasDog: Blank() }},
+  { First: "Helio", Age: 63, Properties: { Vaccinated: Blank(), HasDog: Blank() }},
+  { First: "Mary", Age: 48, Properties: Blank()},
+  { First: "Titan", Age: 79, Properties: Blank() }),Properties.HasDog, SortOrder.Descending)
+Table({Age:2,First:"Bob",Properties:{HasDog:true,Vaccinated:true}},{Age:5,First:"Alice",Properties:{HasDog:true,Vaccinated:true}},{Age:17,First:"John",Properties:{HasDog:false,Vaccinated:false}},{Age:29,First:"Emily",Properties:{HasDog:Blank(),Vaccinated:true}},{Age:63,First:"Helio",Properties:{HasDog:Blank(),Vaccinated:Blank()}},{Age:48,First:"Mary",Properties:Blank()},{Age:79,First:"Titan",Properties:Blank()})
+
+
+// SORT HIERARCHICAL DATA ON STRING EXPRESSION - ASCENDING ORDER
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/Sort.txt:547]: >TRING EXPRESSION [](http://example.com/codeflow?start=30&length=16)

The comment says that the sorting is done on a string expression, but it is actually done in a boolean expresion (same as two tests before that)

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/Sort.txt [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+
+//Error passed as one of the table records
+>> Sort([-2, -1, 5, 1/0, 2], Value)
+#Error(Kind=Div0)
+
+>> Sort(["a", "b", "D", "x", "J", Error({Kind: 11})], Value)
+#Error(Kind=Validation)
+
+>> Sort([1/0], Blank())
+#Error(Kind=Div0)
```
[the remark this answers is not in the document]
Comment (mitugm14) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/Sort.txt:109]: Fixed the expected result

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/Sort.txt (hunk 1 above)
Comment (mitugm14) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/Sort.txt:113]: Fixed the expected result to the correct one - Table({Value:#Error(Kind=13)})

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/Sort.txt (hunk 2 above)
Comment (mitugm14) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/Sort.txt:547]: Fixed the testcase to be consistent with the caption

**Review by lesaltzm — APPROVED** [review]

**Review by CarlosFigueiraMSFT — DISMISSED** [review]
:shipit:

**Review by CarlosFigueiraMSFT — APPROVED** [review]

**Pull request merged**